source/server/listener_manager_impl.h

```diff
@@ -156,7 +169,7 @@ class ListenerImpl : public Listener,
   }
   Upstream::ClusterManager& clusterManager() override { return parent_.server_.clusterManager(); }
   Event::Dispatcher& dispatcher() override { return parent_.server_.dispatcher(); }
-  DrainManager& drainManager() override { return parent_.server_.drainManager(); }
+  Network::DrainDecision& drainManager() override { return *this; }
```

Comment:
I thought about renaming but then at the time I thought it was a breaking change, but now I realize that the while factory context thing is new in 1.4.0 so I will change it.